### PR TITLE
feat(ia): Newspack > Settings > Social - part 2

### PR DESCRIPTION
**Pull request by jaredrethman**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Following from #3257, the PR introduces the X (Twitter) and Meta Pixels for Newspack > Settings > Social tab. This PR also introduces `WizardsToggleHeaderCard`:
* Easily create these specific action card layouts.
* Render props to allow consuming components to act on state values and use state callbacks.
* Abstract and unify common toggle on/off logic into reusable component.

**Additional:**
* Updated Twitter to X.
* Fixed an issue where Meta Pixel IDs would strip leading zeros during route validation callback.

### How to test the changes in this Pull Request:

1. Checkout this code and build assets `git checkout origin/feat/ia-setting-social__pixel && npm run build`
2. Navigate to _/wp-admin/admin.php?page=newspack-settings#/social_
3. Observe the below loads/exists:

Screenshot 2024-07-25 at 09 07 11

Once loaded - test the following scenarios:
* Ensure Toggling on/off works as expected, should show/hide form.
* Test Pixel ID inputs i.e. add a value and click "Save Settings":
   * Meta Pixel ID should only accept numeric values.
   * X Pixel ID should only accept alpha-numeric values.
   * Neither inputs should allow empty values.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/newspack/views/settings/social/meta-pixel.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+					{
+						callback: 'isIntegerId',
+					},
+				],
+			] }
+			renderProp={ ( { settingsUpdates, setSettingsUpdates, isFetching } ) => (
+				<TextControl
+					value={ settingsUpdates?.pixel_id ?? '' }
+					label={ __( 'Pixel ID', 'newspack-plugin' ) }
+					type="number"
```
Comment (dkoo) [inline assets/wizards/newspack/views/settings/social/meta-pixel.tsx]: Can Meta pixels contain non-alphanumeric characters? The `type="number"` means that if I enter some other character it shows the `Value cannot be empty!` error:

Screenshot 2024-07-30 at 10 42 33 AM

**Review by dkoo — CHANGES_REQUESTED** [review]
Looking good! Just one question below.

File: assets/wizards/newspack/views/settings/social/meta-pixel.tsx (hunk 1 above)
Comment (jaredrethman) [reply assets/wizards/newspack/views/settings/social/meta-pixel.tsx]: Well spotted @dkoo , I have removed the `type` attribute

**Review by dkoo — APPROVED** [review]
Working well!

**Pull request merged**